src/main/java/com/cleanroommc/groovyscript/compat/mods/immersivepetroleum/Distillation.java

```diff
@@ -0,0 +1,171 @@
+            @Example(value = "fluid('lubricant')", commented = true)
+    })
+    public void removeByOutput(IIngredient output) {
+        DistillationRecipe.recipeList.removeIf(r -> {
+            for (ItemStack itemstack : r.getItemOutputs()) {
+                if (output.test(itemstack)) {
+                    addBackup(r);
+                    return true;
+                }
+            }
+            for (FluidStack fluidStack : r.getFluidOutputs()) {
+                if (output.test(fluidStack)) {
+                    addBackup(r);
+                    return true;
+                }
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(example = @Example("fluid('oil')"))
+    public void removeByInput(IIngredient input) {
+        DistillationRecipe.recipeList.removeIf(r -> {
+            for (FluidStack fluidStack : r.getFluidInputs()) {
+                if (input.test(fluidStack)) {
+                    addBackup(r);
+                    return true;
+                }
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(type = MethodDescription.Type.QUERY)
+    public SimpleObjectStream<DistillationRecipe> streamRecipes() {
+        return new SimpleObjectStream<>(DistillationRecipe.recipeList).setRemover(this::remove);
+    }
+
+    @MethodDescription(priority = 2000, example = @Example(commented = true))
+    public void removeAll() {
+        DistillationRecipe.recipeList.forEach(this::addBackup);
+        DistillationRecipe.recipeList.clear();
+    }
+
+    @Property(property = "fluidInput", valid = @Comp("1"))
+    @Property(property = "output", valid = {@Comp(type = Comp.Type.GTE, value = "0"), @Comp(type = Comp.Type.LTE, value = "Integer.MAX_VALUE")})
+    @Property(property = "fluidOutput", valid = {@Comp(type = Comp.Type.GTE, value = "0"), @Comp(type = Comp.Type.LTE, value = "Integer.MAX_VALUE")})
+    public static class RecipeBuilder extends AbstractRecipeBuilder<DistillationRecipe> {
+
+        @Property(valid = @Comp(value = "0", type = Comp.Type.GTE))
+        private final FloatArrayList chance = new FloatArrayList();
+        @Property(valid = @Comp(value = "0", type = Comp.Type.GTE))
+        private int time;
+        @Property(valid = @Comp(value = "0", type = Comp.Type.GTE))
+        private int energy;
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder chance(float chance) {
+            this.chance.add(chance);
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder chance(float... chances) {
+            for (float chance : chances) {
+                chance(chance);
+            }
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder chance(Collection<Float> chances) {
+            for (float chance : chances) {
+                chance(chance);
+            }
+            return this;
+        }
```

Comment:
Are these chances for specific output? If yes, its not a good idea to do it like this. Instead have a `input(IIngredient, float)` method